pkg/event/controller.go

```diff
@@ -95,9 +113,30 @@ func (gen *generator) Run(ctx context.Context, workers int, waitGroup *sync.Wait
 }
 
 func (gen *generator) startRecorders(ctx context.Context) error {
-	if err := gen.broadcaster.StartRecordingToSinkWithContext(ctx); err != nil {
+	eventHandler := func(obj runtime.Object) {
+		event, ok := obj.(*apieventsv1.Event)
+		if !ok {
+			gen.logger.Error(nil, "unexpected type, expected eventsv1.Event")
+			return
+		}
+
+		eventCopy := event.DeepCopy()
+		eventCopy.ResourceVersion = ""
+		_, err := gen.eventsClient.Events(event.Namespace).Create(ctx, eventCopy, metav1.CreateOptions{})
+		if err != nil {
+			gen.droppedEventsCounter.Add(ctx, 1)
+			gen.logger.Error(err, "failed to create event")
+		}
+	}
+	stopWatcher, err := gen.broadcaster.StartEventWatcher(eventHandler)
+	if err != nil {
 		return err
 	}
+	go func() {
+		<-ctx.Done()
+		stopWatcher()
+	}()
+
```

Comment:
There's no guarantee that this will be called.

Reply:
Can you elaborate more please. From what I understand I wait for the context to be done and then stop the watcher. Do you mean that the context somehow could not gracefully shut?

Reply:
Fixed